### ISSUE TITLE
Rule Change: `no-control-regex` should check Unicode code point escapes

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#15809]
### What rule do you want to change?

no-control-regex

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint no-control-regex: "error"*/

/\u001F/;

/\u{1F}/u;
```

### What does the rule currently do for this code?

Reports only `/\u001F/`.

### What will the rule do after it's changed?

Report both `/\u001F/` and `/\u{1F}/u`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

I think not reporting `/\u{1F}/u` is actually a bug caused by forgetting to pass the `u` flag to `regexpp`, so `\u{1F}` is not parsed as an escape sequence. The bug is observable in (rare) patterns that are valid with the `u` flag but invalid without it. For example, the rule fails to report `\x1F` in `/\u{1111}*\x1F/u` because `regexpp` stops parsing the pattern after `\u{1111}*` since it is invalid without the `u` flag.

Here's a demo with examples: [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tY29udHJvbC1yZWdleDogXCJlcnJvclwiKi9cblxuL1xcdTAwMUYvOyAvLyB0aGlzIGlzIHJlcG9ydGVkXG5cbi9cXHV7MUZ9L3U7IC8vIHRoZW4gd2h5IG5vdCB0aGlzIGFzIHdlbGw/XG5cbi8vLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS1cblxuL1xceDFGLzsgLy8gdGhpcyBpcyByZXBvcnRlZFxuXG4vXFx1ezExMTF9KlxceDFGL3U7IC8vIGJ1dCB0aGlzIGlzbid0IGFsdGhvdWdoIGl0IGhhcyB0aGUgc2FtZSBgXFx4MUZgIHNlcXVlbmNlXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by nzakas** [comment]
Makes sense to me.

**Comment by nzakas** [comment]
@btmills ?

**Comment by btmills** [comment]
Makes sense to me as well as a semver-minor bug fix :+1:
